mmdet/datasets/dataset_wrappers.py

```diff
@@ -51,5 +68,86 @@ def __init__(self, dataset, times):
     def __getitem__(self, idx):
         return self.dataset[idx % self._ori_len]
 
+    def get_cat_ids(self, idx):
+        self.dataset.get_cat_ids(idx % self._ori_len)
+
     def __len__(self):
         return self.times * self._ori_len
+
+
+# Modified from https://github.com/facebookresearch/detectron2/blob/41d475b75a230221e21d9cac5d69655e3415e3a4/detectron2/data/samplers/distributed_sampler.py#L57 # noqa
+@DATASETS.register_module()
+class RepeatFactorDataset(object):
+    """A wrapper of repeated dataset with repeat factor.
+
+    Suitable for training on class imbalanced datasets like LVIS. Following
```

Comment:
The way that enables all datasets could be encapsulated by RepeatFactorDataset is to implement the function `get_cat_ids`, maybe the docstring should point out this as well as other related documentation (e.g., tutorials about dataset usage.